### PR TITLE
IllegalArgumentException when calling RxHelper.deployVerticle (RxJava2)

### LINKED ISSUE #224: Support nested method type arguments
The generator supports a limited form or type argument reification, e.g `ReadStream<Message<T>>` which was sufficient as the design of API was not requiring more.

The recent fluent query API for SQL client introduces an addition of nesting that the generator does not support leading to incorrect wrapping leading to `ClassCastException` (see https://github.com/vert-x3/vertx-rx/issues/223).

We shall support such types in order to use the new fluent query API with the rx-ified API, i.e with types declarations such as `Query<RowSet<Row>>`.

### PR DESCRIPTION
**Pull request by tsegismont**
Fixes vert-x3/vertx-examples#224

Issue not found while porting features from RxJava module

### CODE REVIEW

File: rx-java2/src/main/java/io/vertx/reactivex/core/RxHelper.java [hunk 1]
```diff
@@ -56,9 +56,15 @@ public static Scheduler blockingScheduler(Vertx vertx, boolean ordered) {
    * @return the response observable
    */
   public static Single<String> deployVerticle(Vertx vertx, Verticle verticle, DeploymentOptions options) {
-    return Single.unsafeCreate(new io.vertx.reactivex.core.impl.AsyncResultSingle<String>(handler -> {
-      vertx.getDelegate().deployVerticle(verticle, options, handler);
-    }));
+    return Single.create(emitter -> {
```
Comment (vietj) [inline rx-java2/src/main/java/io/vertx/reactivex/core/RxHelper.java]: what does it change to use this instead of the previous construct ?

Comment (tsegismont) [reply rx-java2/src/main/java/io/vertx/reactivex/core/RxHelper.java]: It's the default choice for bridging callback/reactive code. RxJava adds some safety checks with respect to threading and also manages cancellation. You only have to deal with `onError` and `onSuccess`.

Comment (vietj) [reply rx-java2/src/main/java/io/vertx/reactivex/core/RxHelper.java]: why then not use instead:

```
  public static Single<String> deployVerticle(Vertx vertx, Verticle verticle, DeploymentOptions options) {
    return new io.vertx.reactivex.core.impl.AsyncResultSingle<String>(handler -> {
      vertx.getDelegate().deployVerticle(verticle, options, handler);
    });
  }
```

Comment (tsegismont) [reply rx-java2/src/main/java/io/vertx/reactivex/core/RxHelper.java]: I had not thought about it. I would do just as well, I think (its code more or less does what `io.reactivex.internal.operators.single.SingleCreate` does). I can change the impl for consistency if you want.

Comment (vietj) [reply rx-java2/src/main/java/io/vertx/reactivex/core/RxHelper.java]: yes I would rather use the same scheme, alors it's bit less code

Comment (tsegismont) [reply rx-java2/src/main/java/io/vertx/reactivex/core/RxHelper.java]: Done

**Pull request merged**